pkg/engine/engine_test.go

```diff
@@ -445,4 +445,33 @@ func TestAlterFuncMap(t *testing.T) {
 		t.Errorf("Expected %q, got %q (%v)", expectNum, gotNum, outReq)
 	}
 
+	tplChart := &chart.Chart{
+		Metadata: &chart.Metadata{Name: "TplFunction"},
+		Templates: []*chart.Template{
+			{Name: "templates/base", Data: []byte(`Evaluate tpl {{tpl "Value: {{ .Values.value}}" .}}`)},
```

Comment:
While testing the "include" function I ran into the following error since the other template is only available for the outer render context. Is there an global render context I could use to pass all registered templates to the rendering of the "tpl"-function? 

```
tplChartWithInclude := &chart.Chart{
		Metadata: &chart.Metadata{Name: "TplFunction"},
		Templates: []*chart.Template{
			{Name: "templates/base", Data: []byte(`{{ tpl "{{include `+"`"+`TplFunction/templates/_partial`+ "`" + ` . | indent 2}}" .}}`)},
			{Name: "templates/_partial", Data: []byte(`{{.Release.Name}} - he`)},
		},
		Values:       &chart.Config{Raw: ``},
		Dependencies: []*chart.Chart{},
```

results in: 

`Expected "Evaluate tpl bar", got "  template: no template \"TplFunction/templates/_partial\" associated with template \"gotpl\"" (map[TplFunction/templates/_partial:TestRelease - he TplFunction/templates/base:  template: no template "TplFunction/templates/_partial" associated with template "gotpl"])`